manic/sourcetree.py

```diff
@@ -331,11 +339,23 @@ def checkout(self, verbosity, load_all, load_comp=None):
             printlog('Checking out externals: ', end='')
 
         if load_all:
-            load_comps = self._all_components.keys()
+            tmp_comps = self._all_components.keys()
         elif load_comp is not None:
-            load_comps = [load_comp]
+            tmp_comps = [load_comp]
         else:
-            load_comps = self._required_compnames
+            tmp_comps = self._required_compnames
+
+        load_comps = []
+        for comp in tmp_comps:
+            prereq = self._all_components[comp].get_prereq()
+            if prereq:
+                if prereq in self._all_components and prereq not in load_comps:
+                    load_comps.append(prereq)
+                else:
+                    fatal_error("prereq {} of component {} not found".format(prereq, comp))
```

Comment:
This `else` can be hit in two cases:

1. if prereq is not in self._all_components

2. if prereq is already in load_comps

If I understand this correctly, the error should only be raised due to condition (1), not due to condition (2)